### PR TITLE
RavenDB-12025 Fix NRE and InvalidCast

### PR DESCRIPTION
**Pull request by karmeli87**

### CODE REVIEW

File: src/Raven.Server/ServerWide/Maintenance/ClusterObserver.cs [hunk 1]
```diff
@@ -1177,7 +1177,10 @@ public bool TryGetAutoIndex(string name,out AutoIndexDefinition definition)
                 var deletionInProgress = new Dictionary<string, DeletionInProgressStatus>();
                 foreach (var prop in obj.GetPropertyNames())
                 {
-                    deletionInProgress[prop] = (DeletionInProgressStatus)obj[prop];
+                    if (Enum.TryParse(obj[prop].ToString(), out DeletionInProgressStatus result))
```
Comment (ayende) [inline src/Raven.Server/ServerWide/Maintenance/ClusterObserver.cs:1180]: Is this always going to be a string here?

Comment (karmeli87) [reply src/Raven.Server/ServerWide/Maintenance/ClusterObserver.cs:1180]: Something should go very wrong if this isn't the case.

Comment (ppekrol) [reply src/Raven.Server/ServerWide/Maintenance/ClusterObserver.cs:1180]: Yes, this is from:
```
public Dictionary<string, DeletionInProgressStatus> DeletionInProgress;
```

Comment (ppekrol) [reply src/Raven.Server/ServerWide/Maintenance/ClusterObserver.cs:1180]: @karmeli87 

```
obj[prop]?.ToString()
```

Maybe just in case change to that?

**Pull request merged**